pkg/balancer/least_connections.go

```diff
@@ -0,0 +1,129 @@
+package balancer
+
+import (
+	"math"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/square/p2/pkg/util"
+)
+
+type lcBackend struct {
+	conns int
+
+	// disabling allows us to gracefully release connections without assigning
+	// new connections to the backend.
+	disabled bool
```

Comment:
I think disabling a connection record without removing its connection count is the right way to handle removing an address, but I don't understand what you mean by enabling/disabling allowing for a "graceful release" of connections. Could you clarify?